openpgp/internal/encoding/octet_stream.go

```diff
@@ -0,0 +1,61 @@
+package encoding
+
+import (
+	"io"
+
+	"github.com/ProtonMail/go-crypto/openpgp/errors"
+)
+
+type OctetString struct {
+	length	uint16
+	data []byte
+}
+
+func NewOctetString(data []byte) (*OctetString, error) {
+	byteLength := len(data)
+	if byteLength > 65535 {
+		return nil, errors.InvalidArgumentError("Data too long")
```

Comment:
Tbh I think it's acceptable to panic here, or just remove this check entirely and assume that the len fits in a `uint16`, as `NewMPI` does the same and it should never happen that it's too long. That way you don't have to return an error.